xtra-libp2p/src/dialer.rs

```diff
@@ -98,6 +98,17 @@ impl Actor {
 
 #[xtra_productivity]
 impl Actor {
+    async fn handle(&mut self, _msg: StopIfNotConnected, ctx: &mut xtra::Context<Self>) {
+        if !self.connected {
+            self.stop_with_error(
+                Error::Failed {
+                    source: anyhow!("Did not connect in time"),
+                },
+                ctx,
+            )
+        }
+    }
```

Comment:
It's unfortunate that we have to do this, but I can't come up with a better alternative right now. This explains why `connected` was introduced in the earlier commit (although it wasn't needed until now).

I guess it's the nature of pushing as opposed to polling. Perhaps a less invasive solution could be to poll the endpoint in `started` directly until `ConnectionStats` holds the peer ID we expect. That should allow us to remove `connected` and this handler.

Reply:
I suspect it would be the same level of invasiveness and don't see much problem in tracking the state here based on the nofications?

Reply:
To me it seems more natural, predictable and easier to understand to block the startup of the actor until we get feedback from the endpoint that we have a connection.

It would just be one `await` with a timeout.

But not a big deal.